lnwallet/channel_test.go

```diff
@@ -8038,18 +8074,33 @@ func TestChannelMaxFeeRate(t *testing.T) {
 	}
 	defer cleanUp()
 
-	// Anchor commitments are heavier, hence will the same allocation lead
+	// Anchor commitments are heavier, hence the same allocation lead
 	// to slightly lower fee rates.
-	assertMaxFeeRate(
-		anchorChannel, 1.0, chainfee.FeePerKwFloor,
+	assertIdealFeeRate(
+		anchorChannel, 700000000, 0, chainfee.FeePerKwFloor,
+		0.1, chainfee.FeePerKwFloor,
+	)
+
+	// If the maximum fee is lower than the network fee, use the maximum fee
+	assertIdealFeeRate(anchorChannel, 700000000, 0, 1000000, 0.001, 444839)
+	assertIdealFeeRate(anchorChannel, 700000000, 0, 1000000, 0.001, 444839)
+	assertIdealFeeRate(anchorChannel, 700000000, 0, 300000, 0.001, 300000)
+	assertIdealFeeRate(anchorChannel, 700000000, 0, 700, 0.000001, 444)
+	assertIdealFeeRate(
+		anchorChannel, 700000000, 0, 1000000, 0.0000001,
 		chainfee.FeePerKwFloor,
 	)
-	assertMaxFeeRate(anchorChannel, 0.001, 1000000, 444839)
-	assertMaxFeeRate(anchorChannel, 0.001, 300000, 300000)
-	assertMaxFeeRate(anchorChannel, 0.000001, 700, 444)
-	assertMaxFeeRate(
-		anchorChannel, 0.0000001, 1000000, chainfee.FeePerKwFloor,
-	)
+
+	// If the maximum anchor commit fee rate is less than the maximum fee,
+	// use the max anchor commit fee since this is an anchor channel.
+	assertIdealFeeRate(anchorChannel, 700000000, 0, 400000, 0.001, 400000)
+
+	// If the minimum relay fee is above the max anchor commitment fee,
+	// use the min relay fee.
+	assertIdealFeeRate(anchorChannel, 700000000, 400000, 300000, 0.001, 400000)
+
+	// If the min relay fee is above the maximum fee, use the maximum fee.
```

Comment:
I think we should override the maximum fee if it's under the min relay fee so the commitment tx has a chance of getting confirmed